org.jcryptool.core.introduction/src/org/jcryptool/core/introduction/views/AlgorithmInstruction.java

```diff
@@ -260,27 +264,30 @@ public void createPartControl(Composite parent) {
 		gl_content.horizontalSpacing = 0;
 		gl_content.verticalSpacing = 0;
 		content.setLayout(gl_content);
+		content.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 //		content.setBackground(ColorService.GREEN);
 		
 		scrolledComposite.setContent(content);
 
 
 		initializeScaledImages();
 		
-		
-		// Composite that should grab space on the left.
-		// This should center the slideshow
-		Composite left = new Composite(content, SWT.NONE);
-		GridData gd_left = new GridData(SWT.FILL, SWT.FILL, false, false);
-		left.setLayoutData(gd_left);
-		GridLayout gl_left = new GridLayout();
-		gl_left.marginHeight = 0;
-		gl_left.marginWidth = 0;
-		left.setLayout(gl_left);
+		content.addListener(SWT.Resize, new Listener() {
+			@Override
+			public void handleEvent(Event event) {
+				int[] sizehint = computeSlideshowSizeHint();
+				gridData_cnvs.widthHint = sizehint[0];
+				gridData_cnvs.heightHint = sizehint[1];
+				content.layout(new Control[] {cnvs});
+			}
+		});
 		
 		// The canvas the slideshow is painted on.
 		cnvs = new Canvas(content, SWT.DOUBLE_BUFFERED);
-		gridData_cnvs = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gridData_cnvs = new GridData(SWT.CENTER, SWT.FILL, true, true);
```

Comment:
This part is crucial for centering:
1) the space to center in must be actually there (see above comment -- make the parent composite have some room available)
2) the space must be actually grabbed to consider "centering in it"
3) the alignment must be `SWT.CENTER`, of course.

*adjust the vertical alignment to `SWT.CENTER` if needed"*